src/Diagnostics.DataProviders/TokenService/LogAnalyticsTokenServiceBase.cs

```diff
@@ -84,7 +84,7 @@ private async Task Authenticate()
                 ValidateAuthority = true
             };
 
-            creds = await ApplicationTokenProvider.LoginSilentAsync(domain, clientId, clientSecret, adSettings);
+            creds = await ApplicationTokenProvider.LoginSilentAsync(domain, clientId, clientSecret, adSettings).ConfigureAwait(false);
```

Comment:
Do we need `ConfigureAwait(false)` here?

Reply:
Michi so this part of this code is not run as background task but the StartTokenRefresh method which is being called somewhere else in runtimehost code might be better suited to have the `ConfigureAwait(false)` as that is run as a background task.

Reply:
https://github.com/Azure/Azure-AppServices-Diagnostics/pull/786/files#diff-158490db55e5872149db5f22ad70d157ac7306decd376acb31ea58a33061daffR39